File: ddb/sdk/typescript/examples/inspect.ts
import { newClient } from "./config.js";

const client = newClient();
try {
  const { serverInfo, capabilities } = await client.handshake();
  const response = await client.call("DebuggerService.GetSnapshot", {
    sections: [
      "SNAPSHOT_SECTION_TOPOLOGY",
      "SNAPSHOT_SECTION_SELECTION",
      "SNAPSHOT_SECTION_EXECUTION",
    ],
  });
  console.log(serverInfo.version, capabilities.schemaVersion);
  for (const thread of response.snapshot?.threads ?? []) {
    console.log(thread.threadId, thread.state, thread.location);
  }
} finally {
  client.close();
}

File: ddb/sdk/python/examples/inspect.py
from _config import new_client


with new_client() as client:
    server, capabilities = client.handshake()
    response = client.get_snapshot(
        {
            "sections": [
                "SNAPSHOT_SECTION_TOPOLOGY",
                "SNAPSHOT_SECTION_SELECTION",
                "SNAPSHOT_SECTION_EXECUTION",
            ]
        }
    )
    print(server.get("version"), capabilities.get("schemaVersion"))
    for thread in response.get("snapshot", {}).get("threads", []):
        print(thread.get("threadId"), thread.get("state"), thread.get("location"))
